src/model/CollectionNode.ts

```diff
@@ -92,7 +99,7 @@ export default class CollectionNode implements INode {
           );
           logger.info(`Created Primay Index on ${this.bucketName} ${this.scopeName} ${this.collectionName} USING GSI`);
           result = await this.connection.cluster?.query(
-            `SELECT RAW META().id FROM \`${this.bucketName}\`.\`${this.scopeName}\`.\`${this.collectionName}\` LIMIT ${this.limit}`
+            `SELECT RAW META().id FROM \`${this.bucketName}\`.\`${this.scopeName}\`.\`${this.collectionName}\` ${filter.length>0 ? "WHERE "+filter: ""} LIMIT ${this.limit}`
           );
         }
```

Comment:
Can you look into the filter with 
name = "Rigoberto Bernier" on tenant_agent_0 user